### PR TITLE
Adding mq | tag remove-tag mark-for-op 

### PR DESCRIPTION
**Pull request by PratMis**

### CODE REVIEW

File: c7n/resources/mq.py [hunk 1]
```diff
@@ -74,3 +84,109 @@ def process(self, resources):
...
+                    key: target-tag
+                    value: target-tag-value
+    """
+
+    permissions = ('mq:TagMessageBroker')
+
+    def process_resource_set(self, mq, new_tags):
+        client = local_session(self.manager.session_factory).client('mq')
+        tag_dict = {}
+        for t in new_tags:
```
Comment (kapilt) [inline c7n/resources/mq.py]: these three lines could be simplified to a one liner re dict comprehension.

**Comment by PratMis** [comment]
Thanks! Got it, will put in the requested changes

**Comment by PratMis** [comment]
@kapilt, not sure why tox is still failing! Looks like there's some package mismatch. Do you want me to open a fresh PR? Thank you!

File: c7n/resources/mq.py [hunk 2]
```diff
@@ -74,3 +86,96 @@ def process(self, resources):
...
+                resource: message-broker
+                filters:
+                  - "tag:target-tag": absent
+                actions:
+                  - type: tag
+                    key: target-tag
+                    value: target-tag-value
+    """
+
+    permissions = ('mq:TagMessageBroker')
```
Comment (kapilt) [inline c7n/resources/mq.py]: needs a trailing comma

File: c7n/resources/mq.py [hunk 3]
```diff
@@ -74,3 +86,96 @@ def process(self, resources):
...
+              - name: mq-remove-tag
+                resource: message-broker
+                filters:
+                  - "tag:OutdatedTag": present
+                actions:
+                  - type: remove-tag
+                    tags: ["OutdatedTag"]
+    """
+
+    permissions = ('mq:UntagMessageBroker')
```
Comment (kapilt) [inline c7n/resources/mq.py]: needs a trailing comma

File: c7n/resources/mq.py [hunk 4]
```diff
@@ -74,3 +86,96 @@ def process(self, resources):
...
+                  - "tag:custodian_cleanup": absent
+                actions:
+                  - type: mark-for-op
+                    tag: custodian_cleanup
+                    msg: "Unused mq"
+                    op: delete
+                    days: 7
+    """
+
+    permissions = ('mq:TagMessageBroker')
```
Comment (kapilt) [inline c7n/resources/mq.py]: needs a trailing comma

**Comment by PratMis** [comment]
Thanks, just pushed those changes in

File: c7n/resources/mq.py [hunk 5]
```diff
@@ -37,6 +38,17 @@ class resource_type(object):
         filter_name = None
         metrics_namespace = 'AWS/AmazonMQ'
 
+    def augment(self, resources):
+        super(MessageBroker, self).augment(resources)
+        client = local_session(self.session_factory).client('mq')
+        for r in resources:
+            try:
+                tags = self.retry(client.list_tags, ResourceArn=r['BrokerArn'])['Tags']
```
Comment (kapilt) [inline c7n/resources/mq.py]: why is this being done? tags are already in describe outputs, please remove as extraneous.

Comment (PratMis) [reply c7n/resources/mq.py]: The tags are given out in the form of a dictionary. This was done to switch them to a list form

File: tests/test_mq.py [hunk 6]
```diff
@@ -67,11 +67,61 @@ def test_delete_mq(self):
...
+                     'days': 2}]},
+            config={'region': 'us-east-1'},
+            session_factory=factory,
+        )
+        resources = p.run()
+        self.assertTrue(len(resources), 1)
+        client = factory().client("mq")
+        if self.recording:
+            time.sleep(1)
+        tags = client.list_tags(ResourceArn=resources[0]["BrokerArn"])["Tags"]
```
Comment (kapilt) [inline tests/test_mq.py]: the tags here are never used in an assertion.

File: c7n/resources/mq.py (hunk 5 above)
Comment (kapilt) [reply c7n/resources/mq.py]: why are tags being fetched from the api in a separate call per resource, when they are already present in the describe output?

Comment (PratMis) [reply c7n/resources/mq.py]: Understood! Thanks

**Comment by kapilt** [comment]
i went ahead and cleaned up the test and the extraneous changes to recorded files to get a minimal diff. lgtm, thanks.

**Pull request merged**